src/task2.1/index.js

```diff
@@ -2,12 +2,27 @@ import express from 'express';
 
 import routes from './routes';
 import errorMiddleware from './middlewares/error-middleware';
+import loggerMiddleware from './middlewares/logger-middleware';
+import winstonLogger from './middlewares/winston-logger';
 
 const app = express();
 
 app.listen(process.env.PORT || 8080);
 
-app.use(express.urlencoded({ extended: true }));
 app.use(express.json());
-app.use(routes);
+
+app.use(express.urlencoded({ extended: true }));
+
+app.use(loggerMiddleware);
 app.use(errorMiddleware);
+
+app.use(routes);
```

Comment:
the error middleware is in use before your routes. So, it will not handle any error that you intend to pass there.